src/main/java/seedu/address/logic/parser/CreateGroupParser.java

```diff
@@ -0,0 +1,136 @@
+        ReadOnlyAddressBook addressBook = model.getAddressBook();
+        this.allMembers = addressBook.getPersonList();
+        this.toBeAddedToGroup = new ArrayList<>();
+    }
+    //assume command is of the following form
+    //creategroup Berlin : Chrus Chong : Nancy Chua : Tom Hiddleston
+
+    /**
+     * Returns CreateGroupCommand based on user input.
+     *
+     * @param args User input into addressbook.
+     * @return CreateGroupCommand object to represent command to be executed.
+     * @throws ParseException If user input is incorrectly formatted.
+     */
+    public CreateGroupCommand parse(String args) throws ParseException {
+        GroupName groupName = new GroupName(findGroupName(args));
+        ArrayList<Person> members = findGroupMembers(args);
+        boolean validCommand = true;
+        if (groupName.equals(BAD_FORMATTING) || Objects.isNull(members)) {
+            validCommand = false;
+        }
+        return new CreateGroupCommand(groupName, members, validCommand);
+    }
+
+    /**
+     * Returns group name from user input.
+     * If user input invalid, BAD_FORMATTING is returned.
+     *
+     * @param args String object representing user input into the addressbook.
+     * @return String object representing group name.
+     */
+    private String findGroupName(String args) {
+        //assuming it is already a valid command which will be checked beforehand
+        try {
+            int startIndex = 1;
+            int endIndex = args.indexOf(':') - 1;
+            return args.substring(startIndex, endIndex);
+        } catch (IndexOutOfBoundsException e) {
+            return BAD_FORMATTING;
+        }
+    }
+
+    /**
+     * Returns list of members in the group.
+     * If user input not valid, null is returned.
+     *
+     * @param args String object representing user input into the addressbook.
+     * @return ArrayList of Person objects representing members to be added to the group
+     */
+    private ArrayList<Person> findGroupMembers(String args) {
+        try {
+            int startIndex = args.indexOf(':') + 2;
+            String teamMembers = args.substring(startIndex);
+
+            int nextColon = teamMembers.indexOf(':');
+            while (nextColon != -1) {
+                Name memberName = new Name(teamMembers.substring(0, nextColon - 1));
+                if (Objects.isNull(addMemberIfExist(memberName))) {
+                    return null;
+                }
+                teamMembers = teamMembers.substring(nextColon + 2);
+                nextColon = teamMembers.indexOf(':');
+            }
+            if (Objects.isNull(addMemberIfExist(new Name(teamMembers)))) {
+                return null;
+            }
+            if (toBeAddedToGroup.size() == 0) {
+                return null;
+            } else {
+                return toBeAddedToGroup;
+            }
+        } catch (IndexOutOfBoundsException e) {
+            return null;
+        }
+
+    }
```

Comment:
Perhaps it can return optional instead of null?

Reply:
@marcuspeh this was really sharp of you to notice. Thank you very much.